tests/rptest/tests/schema_registry_test.py

```diff
@@ -1223,6 +1223,122 @@ def move_leader():
+                self.logger.debug(res)
+                assert "id" in res
+                schema_ids.append(result_raw.json()["id"])
+
+            return schemas, schema_ids
+
+        # Given a list of subjects, check that they exist on the registry
+        def check_subjects(subjects: list[str]):
+            result_raw = self._get_subjects()
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
+            res_subjects = result_raw.json()
+            self.logger.debug(res_subjects)
+            assert type(res_subjects) == type([])
+            res_subjects.sort()
+            subjects.sort()
+            assert res_subjects == subjects
+
+        # Given the subject and list of ids, check that the ids match
+        def check_subject_versions(subject: str, schema_ids: list[int]):
+            result_raw = self._get_subjects_subject_versions(subject=subject)
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
+            res_ids = result_raw.json()
+            self.logger.debug(res_ids)
+            assert type(res_ids) == type([])
+            res_ids.sort()
+            schema_ids.sort()
+            assert res_ids == schema_ids
+
+        # Given the subject, list of schemas, and list of ids, check the schema that maps to the id
+        def check_each_schema(subject: str, schemas: list[str],
+                              schema_ids: list[int]):
+            for schema_id in schema_ids:
+                self.logger.debug(
+                    f"Fetch schema {schema_id} on subject {subject}")
+                result_raw = self._get_subjects_subject_versions_version(
+                    subject=subject, version=schema_id)
+                self.logger.debug(result_raw)
+                assert result_raw.status_code == requests.codes.ok
+                res = result_raw.json()
+                self.logger.debug(res)
+                assert type(res) == type({})
+                assert "subject" in res
+                assert "version" in res
+                assert "id" in res
+                assert "schema" in res
+                assert res["subject"] == subject
+                assert res["version"] == schema_id
+                assert res["id"] == schema_id
+                schema = json.loads(schemas[schema_id - 1])
+                assert res["schema"] == schema["schema"]
+
+        admin = Admin(self.redpanda)
+
+        # Failure checks
+        try:
+            admin.redpanda_services_restart()
+        except requests.exceptions.HTTPError as ex:
+            self.logger.debug(ex)
+            assert ex.response.status_code == requests.codes.bad_request
+
+        try:
+            admin.redpanda_services_restart(rp_service='foobar')
+        except requests.exceptions.HTTPError as ex:
+            self.logger.debug(ex)
+            assert ex.response.status_code == requests.codes.not_found
```

Comment:
These failure tests are more for the restart admin endpoint than for schema registry.

Reply:
Fair point, I'll find another place for these then.